src/Processors/Formats/IInputFormat.cpp

```diff
@@ -16,7 +16,10 @@ Chunk IInputFormat::generate()
 {
     try
     {
-        return read();
+        auto chunk = read();
+        chunk.getChunkInfos().add(std::make_shared<ChunkInfoReadRowsBefore>(total_rows));
```

Comment:
IFormat can call another IFormat::read and in this case result chunk will recieve two instances of ChunkInfoReadRowsBefore. You should check the existence of ChunkInfoReadRowsBefore before adding